File: kata-5-kyu/paginationhelper/python/paginationhelper_test.py
# + =====================================================================================================================
# +
# + 5 kyu - PaginationHelper  [ ID: 515bb423de843ea99400000a ] (paginationhelper)
# + URL: https://www.codewars.com/kata/515bb423de843ea99400000a
# + Category: ALGORITHMS  |  Tags: OBJECT-ORIENTED PROGRAMMING | ARRAYS | ALGORITHMS
# +
# + =====================================================================================================================

import codewars_test as test
from paginationhelper import UNKNOWN


test.describe("Basic tests")
collection = range(1, 25)
helper = PaginationHelper(collection, 10)

test.it("Testing .page_count() method")
test.assert_equals(helper.page_count(), 3, "page_count is returning incorrect value.")

test.it("Testing .page_item_count() method")
test.assert_equals(helper.page_item_count(1), 10, "page_item_count is returning incorrect value.")
test.assert_equals(helper.page_item_count(2), 4, "page_item_count is returning incorrect value")
test.assert_equals(helper.page_item_count(3), -1, "page_item_count is returning incorrect value")

test.it("Testing .page_index() method")
test.assert_equals(helper.page_index(0), 0, "page_index returned incorrect value")
test.assert_equals(helper.page_index(23), 2, "page_index returned incorrect value")
test.assert_equals(
    helper.page_index(24),
    -1,
    "page_index returned incorrect value when provided a item_index argument that was out of range",
)
test.assert_equals(
    helper.page_index(40),
    -1,
    "page_index returned incorrect value when provided a item_index argument that was out of range",
)
test.assert_equals(helper.page_index(3), 0, "page_index returned incorrect value")
test.assert_equals(
    helper.page_index(-1),
    -1,
    "page_index returned incorrect value when provided a itemIndex argument that was out of range. pageIndex(-1) should return -1",
)
test.assert_equals(
    helper.page_index(-23),
    -1,
    "page_index returned incorrect value when provided a item_index argument that was out of range. pageIndex(-23) shoudl return -1",
)
test.assert_equals(
    helper.page_index(-15),
    -1,
    "page_index returned incorrect value when provided a item_index argument that was out of range.",
)

test.it("Testing .item_count() method")
test.assert_equals(helper.item_count(), 24, "item_count returned incorrect value")

test.it("Testing empty array")
helper = PaginationHelper([], 10)
test.assert_equals(helper.page_index(0), -1, "pageIndex(0) called when there was an empty collection")

File: kata-5-kyu/paginationhelper/javascript/paginationhelper.Test.js
//+ ====================================================================================================================
//+
//+ 5 kyu - PaginationHelper  [ ID: 515bb423de843ea99400000a ] (paginationhelper)
//+ URL: https://www.codewars.com/kata/515bb423de843ea99400000a
//+ Category: ALGORITHMS  |  Tags: OBJECT-ORIENTED PROGRAMMING | ARRAYS | ALGORITHMS
//+
//+ ====================================================================================================================

import { assert, expect } from "vitest"
const { PaginationHelper } = require("./paginationhelper")

describe("Tests", () => {
  it("test", () => {
    const collection = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24]
    let helper = new PaginationHelper(collection, 10)

    expect(helper.pageCount() == 3, "pageCount is returning incorrect value.")

    expect(helper.pageItemCount(1) == 10, "pageItemCount is returning incorrect value.")
    expect(helper.pageItemCount(2) == 4, "pageItemCount is returning incorrect value")
    expect(helper.pageItemCount(3) == -1, "pageItemCount is returning incorrect value")
    expect(
      helper.pageIndex(40) == -1,
      "pageIndex returned incorrect value when provided a itemIndex argument that was out of range"
    )

    expect(helper.pageIndex(22) == 2, "pageIndex returned incorrect value")
    expect(helper.pageIndex(3) == 0, "pageIndex returned incorrect value")
    expect(helper.pageIndex(0) == 0, "pageIndex returned incorrect value. pageIndex(0) should return 0")
    expect(
      helper.pageIndex(-1) == -1,
      "pageIndex returned incorrect value when provided a itemIndex argument that was out of range. pageIndex(-1) should return -1"
    )
    expect(
      helper.pageIndex(-23) == -1,
      "pageIndex returned incorrect value when provided a item_index argument that was out of range. pageIndex(-23) shoudl return -1"
    )
    expect(
      helper.pageIndex(-15) == -1,
      "pageIndex returned incorrect value when provided a item_index argument that was out of range."
    )

    expect(helper.itemCount() == 24, "itemCount returned incorrect value")

    helper = new PaginationHelper([], 10)
    assert.strictEqual(helper.pageIndex(0), -1, "pageIndex(0) called when there was an empty collection")
  })
})
